lucille-core/src/test/java/com/kmwllc/lucille/stage/CollapseChildrenDocumentsTest.java

```diff
@@ -0,0 +1,49 @@
+package com.kmwllc.lucille.stage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import org.junit.Test;
+
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+
+public class CollapseChildrenDocumentsTest {
+
+  private StageFactory factory = StageFactory.of(CollapseChildrenDocuments.class);
+
+  @Test
+  public void testCollapsedFields() throws StageException {
+    Stage stage = factory.get("CollapseChildrenDocumentsTest/config.conf");
+
+    Document doc = Document.create("doc");
+    doc.setField("parent", true);
+    
+    Document child1 = Document.create("child1");
+    child1.setField("childfield1", "foo");
+    child1.setField("childfield2", "bar");
+    doc.addChild(child1);
+
+    Document child2 = Document.create("child2");
+    child2.setField("childfield1", "foo");
+    child2.setField("childfield2", "bar");
+    child2.setField("childfield3", "baz");
+    doc.addChild(child2);
+
+    stage.processDocument(doc);
+    
+    assertFalse(doc.hasChildren());

```

Comment:
should ideally also have a test where the config has dropChildren: false 
and we make sure the children aren't dropped

Reply:
i added another test to assert that children documents were not dropped.